### DISCUSSION TITLE
EMQX/K8S: can't log in to dashboard

### DISCUSSION THREAD

**Discussion by nguyeny1994** [emqx/emqx#9903] [General]
Hi All,
I tried to create Deployment on k8s by yml file as the below:
apiVersion: v1
kind: Service
metadata:
  name: emqx
  namespace: backend
spec:
  ports:
    - name: mqtt
      nodePort: 31883
      port: 1883
      protocol: TCP
      tartgetPort: 1883
    - name: mqttssl
      nodePort: 30883
      port: 8883
      protocol: TCP
      tartgetPort: 8883
    - name: mgmt
      nodePort: 30081
      port: 8081
      protocol: TCP
      tartgetPort: 8081
    - name: ws
      nodePort: 30083
      port: 8083
      protocol: TCP
      tartgetPort: 8083
    - name: wss
      nodePort: 30084
      port: 8084
      protocol: TCP
      tartgetPort: 8084
    - name: dashboard
      nodePort: 31083
      port: 18083
      protocol: TCP
      tartgetPort: 18083
  selector:
    workload.user.cattle.io/workloadselector: emqx
  type: NodePort
status:
  loadBalancer: {}

---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: emqx
  namespace: backend
spec:
  replicas: 2
  selector:
    matchLabels:
      workload.user.cattle.io/workloadselector: emqx
  template:
    metadata:
      labels:
        workload.user.cattle.io/workloadselector: emqx
    spec:
      imagePullSecrets:
      - name: public
      containers:
      - image: emqx/emqx
        imagePullPolicy: Always
        name: emqx
        ports:
        - containerPort: 1883
          name: mqtt
          protocol: TCP
        - containerPort: 8883
          name: mqttssl
          protocol: TCP
        - containerPort: 8081
          name: mgmt
          protocol: TCP
        - containerPort: 8083
          name: ws
          protocol: TCP
        - containerPort: 8084
          name: wss
          protocol: TCP
        - containerPort: 18083
          name: dashboard
          protocol: TCP
        resources: {}
        restartPolicy: Always
        env:
        - name: EMQX_NAME
          value: emqx
        - name: EMQX_CLUSTER__DISCOVERY
          value: k8s
        - name: EMQX_CLUSTER__K8S__APP_NAME
          value: emqx
        - name: EMQX_CLUSTER__K8S__SERVICE_NAME
          value: emqx
        - name: EMQX_CLUSTER__K8S__APISERVER
          value: "https://kubernetes.default.svc:443"
        - name: EMQX_CLUSTER__K8S__NAMESPACE
          value: backend
When the replicas is 1, I can access to dashboard normally.
But when I scale replicas to 2 or 3, it had the message "Login has expired, please login again" as the below:
![image](https://user-images.githubusercontent.com/86542825/216537993-1ab4ede5-9644-4d69-897e-ddbf3383bcc1.png)
Please support me this issue.
Thank you very much.

**Comment by id** [comment]
You need a load balancer and sticky sessions so that your requests within the session are always routed to the same node.

**Comment by thalesmg** [comment]
Also, try to connect to core nodes when using the dashboard.
